metadata_service/api/column.py

```diff
@@ -3,16 +3,45 @@
 
 import json
 from http import HTTPStatus
-from typing import Iterable, Union
+from typing import Iterable, Mapping, Union
 
+from amundsen_common.models.lineage import LineageSchema
 from flasgger import swag_from
 from flask import request
-from flask_restful import Resource
+from flask_restful import Resource, reqparse
 
+from metadata_service.entity.resource_type import ResourceType
 from metadata_service.exception import NotFoundException
 from metadata_service.proxy import get_proxy_client
 
 
+class ColumnLineageAPI(Resource):
+    """
+    ColumnLineageAPI supports GET operation to get column lineage
+    """
+    def __init__(self) -> None:
+        self.client = get_proxy_client()
+        self.parser = reqparse.RequestParser()
+        self.parser.add_argument('direction', type=str, required=False)
+        self.parser.add_argument('depth', type=int, required=False)
+        super(ColumnLineageAPI, self).__init__()
+
+    @swag_from('swagger_doc/column/lineage_get.yml')
+    def get(self, table_uri: str, column_name: str) -> Iterable[Union[Mapping, int, None]]:
+        args = self.parser.parse_args()
+        direction = args.get('direction', 'both')
+        depth = args.get('depth', 0)
+        try:
+            lineage = self.client.get_lineage(id=f"{table_uri}/{column_name}",
```

Comment:
If this id generation is a common pattern, should it be handled by a util or wrapper?

Reply:
good question, I don't think we have any generic method for both tables and columns, in any methods that call columns like get_column_description on neo4j proxy we get column through its association with table name node, but if we have the entire column key we can just call directly with the key rather than querying neo4j for table->column->what_you_want

Reply:
ah, yeah -- i guess ultimately this is tied to `ColumnMetadata.COLUMN_KEY_FORMAT` -- we're "getting lucky" in that column shares a prefix with `TableMetadata.TABLE_KEY_FORMAT`. I'm sure there are other places that rely on this behavior